test/controllers/deregister/ConfirmStopBeingPsaControllerSpec.scala

```diff
@@ -155,13 +158,13 @@ object ConfirmStopBeingPsaControllerSpec extends ControllerSpecBase {
   private val minimalPsaDetailsNone = MinimalPSA("test@test.com", isPsaSuspended = false, None, None)
   private val minimalPsaDetailsNoneSuspended = MinimalPSA("test@test.com", isPsaSuspended = true, None, None)
 
-  private def fakeAllowAccess(minimalPsaConnector: MinimalPsaConnector): AllowAccessForNonSuspendedUsersActionProvider = {
-    new AllowAccessForNonSuspendedUsersActionProvider {
-      override def apply(): AllowAccessForNonSuspendedUsersAction = new AllowAccessForNonSuspendedUsersAction(minimalPsaConnector)
+  private def fakeAllowAccess(minimalPsaConnector: MinimalPsaConnector): AllowAccessForNonSuspendedUsersAction = {
```

Comment:
Very minimal change needed based on the removal of AllowAccessForNonSuspendedUsersActionProvider, but we should be aiming to use build the app with Guice (with overrides) and retrieve the instance that we need. You don't need to action anything here, just something I think we can bring up with the team

Reply:
Probably right, but part of a wider conversation.